### DISCUSSION TITLE
opentrack 2024.1.1

### DISCUSSION THREAD

**Discussion by sthalik** [opentrack/opentrack#1876] [Announcements]
### Compatibility

The 64-bit build should greatly improve performance for `tracker/neuralnet` users compared to `2023.3.0`. However, `proto/fsuipc` isn't supported anymore and the the legacy CL-Eye driver for the PS3 Eye can't be used. The [open driver](https://github.com/opentrack/opentrack/wiki/PS3-Eye-open-driver-instructions) is recommended.

The 32-bit build doesn't require AVX and can be used by legacy CPUs. However, `tracker/neuralnet` performance is abysmal and actually significantly lower than `2023.3.0`.

### Features

- `filter/accela-hamilton` added as a new filter that combines `filter/hamilton`'s proper rotations and `filter/accela`'s quick response time (by @DaWelter).
- `tracker/neuralnet` now has better performance in this 64-bit build (@DaWelter, @sthalik).
- `core` post-inverting output (i.e. replacing +180 yaw to -180) has been split into separate pre-invert (useful with mirrored camera frames) and post-invert (useful when using output through a virtual joystick or other generic method) (@sthalik).

### Fixes

- `core` fix mapping window taking a large amount of CPU time while tracking is started (@sthalik).
- `core` fix multiple modules such as `tracker/pt` randomly failing to load on Windows 7 or even some Windows 10 machines (please report any remaining issues) (@sthalik).
- `tracker/neuralnet` now works on Windows 7 (see [patch](https://github.com/opentrack/opentrack-depends/blob/master/onnxruntime-win7.patch)) (@sthalik).
- `core` try to fix F13-F23 keystrokes. this might not work or make things worse, please report any bugs (see #1269) (@sthalik).

The debug info package is missing .pdb files for Qt. This should be rectified in the next release.

<hr /><em>This discussion was created from the release <a href='https://github.com/opentrack/opentrack/releases/tag/opentrack-2024.1.1'>opentrack 2024.1.1</a>.</em>

**Comment by qcere** [comment]
Thank you for opentrack, i previously had a facetracking service that had to be payed and still didn't perform as good as opentrack, looking forward for future updates and will there possibly be less delay in the future?

**Comment by Buzoter123** [comment]
Hello!
Thank you very much for the update! Unfortunately, post-invert either doesn’t work or I don’t understand how to use it. Logically, for this example, this flag should give the same approach when turning the head in any direction, but this does not happen - depending on the direction of rotation, the gaze either approaches or moves away. By and large, this flag did not affect anything.
![image](https://github.com/opentrack/opentrack/assets/144796160/04e6b87b-824f-4293-9de4-5ee21c822f3e)

**Reply by sthalik** [reply]
It should work ingame, without showing in the UI. Is that not the case?

**Reply by Buzoter123** [reply]
I checked it in IL2 BOS - the movements of the pilot's head correspond to the movements of the octopus. In other words, it doesn't work correctly.

**Reply by sthalik** [reply]
Added it to the TODO, thanks.

**Comment by Buzoter123** [comment]
I would also like to ask you to add a hot button to disable the X and Y axes. In this case, the gaze should be centered so that it is directed through the sight. After releasing the button, everything should work as usual. This will be very convenient for aiming. Thank you in advance!

**Reply by sthalik** [reply]
There's the "zero" function that disables all axis and sets the view to origin.

**Reply by Buzoter123** [reply]
This is different - when aiming, you often need to be able to rotate your head, but limit the displacement so as not to lose sight.

**Comment by B14Ape** [comment]
Hi there, 
Love the software! I'm using it on FS22 and after only having the standard Tobii Eyetracker 5 program to use, it's nice to use yours and have the ability to lean in the cab (rather than just static looking around). However, when in cab, the view is of the back of the persons head rather than from the eyes, 3rd person instead of 1st person. But the camera setting is set to first person. I've attached a screenshot to show you what I mean.
Do you have and advice to fix this? 
![Farming Simulator 22 Screenshot 2024 10 20 - 23 27 24 44](https://github.com/user-attachments/assets/7934818e-4dc1-40df-9835-3c04d25b4c7a)

**Comment by izolini** [comment]
I noticed that when using racing titles, like Automobilista 2 or Assetto Corsa, it tends to get some jittery while panning slowly, noticeably when you're looking at apex on hairpins. For flights sims, like DCS and IL-2 this is almost unnoticeable, but if you turn your head really slowly, you can feel that the movement is done in very small jumps (take the cockpit structure as reference). 
Trying to figure it out, I've spent 4 hours fiddling with sensitivities/deadzones/filters, and got it almost dialed in, but in the process I found something interesting: 
Whenever I'm into a game (let's say, AC or AMS2), if I get the stuttering, I change the protocol to "use only opantrack, disable trackir", instead of both, and then returning to game it is silk smooth. If I start the game with this option on the profile, then it doesn't work. 
There are tons of people commenting on the internet about this small jittery when panning, so maybe it is worth to check if there's something that could cause this behavior. What exactly happens when I command a change of protocols? 
I'm usually starting with trackir only or both, and then change to opentrack - but sometimes after alt-tabbing to do this I get a black screen returning to the game and have to alt-f4. 
Games are running in full screen, AMD 7800x3d, plenty of FPS. Camera is PS3 with IR filter running trackpoint + delanclip at 56 degrees 320x240x100Hz. Same results with 56 degrees 640x480x75Hz. Also, using the modules/ps3eye-subprocess.exe from 2023 version.

**Reply by DaWelter** [reply]
Sounds familiar. I once had very clear "movement in little jumps" behaviour with the new accelahamilton filter. It was a floating point precision issue. Switching to double precision helped.

There are many other calculations in 32 bit floating point. Perhaps it's similarly not enough for some.

> Whenever I'm into a game (let's say, AC or AMS2), if I get the stuttering, I change the protocol to "use only opantrack, disable trackir", instead of both, and then returning to game it is silk smooth. If I start the game with this option on the profile, then it doesn't work.

This doesn't really fit the hypothesis though. No idea what's going on there.

**Reply by sthalik** [reply]
We could just switch everything back to double again. Splines, filters, PT, Aruco, all of it.

**Reply by izolini** [reply]
> This doesn't really fit the hypothesis though. No idea what's going on there.

Spent five hours today trying other filters and settings, changing usb ports, video drivers, evaluating latency. No avail. The game I used was DCS.
The "changing protocols" isn't working too. I can swear it worked at least twice, but in other game (AMS2), but not in DCS. Sometimes, closing and reopening opentrack helped a bit, but only about a minute or so, with the stuttering worsening as time passed. Videocard was below 100% and memory (VRAM and RAM) wasn't at max either, CPU wasn't pushing either, with plenty of room. 
In the end I changed from accela to accelahamilton and the situation seemed to be better. The stuttering is there, but it is really small compared to what it was.

> Switching to double precision helped.
> We could just switch everything back to double again. Splines, filters, PT, Aruco, all of it.
If switching then solved the isse, it may be worth a try now?

PS: tried with libusb0 drivers as well (currently on K version, as I was in the first report).

**Comment by Buzoter123** [comment]
[IL2-BF109.zip](https://github.com/user-attachments/files/20257137/IL2-BF109.zip)

I want to support the project as much as I can. The attachment contains my opentrack settings, which I have been using for about six months. Unfortunately, at first, the lack of understanding of how to optimally set up a review brought a lot of frustration and unnecessary questions (and advice). Now I have selected settings that make me feel no worse than in VR, although it took getting used to. In general, I'm posting my settings, and I hope that they will be useful to someone.

**Reply by B14Ape** [reply]
Thank you. I too would like to support, and somehow integrate the gaze / eye tracking that the Tobii 5 has into OpenTrack so it does more than just track your head movements. 

I am not at home right now but will have a look at your settings later. Thank you again

**Comment by kartivanavim** [comment]
didnt expect it to work but its actually great

**Reply by Zeti39** [reply]
Do you know when there will be a new version?

**Reply by kartivanavim** [reply]
No but I'm not sure that its needed because it works

**Comment by Pakorro77** [comment]
Is it possible to have a mac os version?
